<?xml version="1.0"?> 
<gazebo version="1.0">
	<world name="default">
		<scene>
			<ambient rgba="1.0 1.0 1.0 1"/>
			<background rgba="0 0 0 1.0">
	<sky material="Gazebo/CloudySky" />
			</background>
			<shadows enabled="true"/>
			<grid enabled="false" />
		</scene>

		<physics type="ode" update_rate="1000">
			<gravity xyz="0 0 -9.81"/>
			<ode>
				<solver type="quick" dt="0.001" iters="1000" sor="1.3"/>
				<constraints cfm="0.0000000001" erp="0.2" contact_max_correcting_vel="100.0" contact_surface_layer="0.001"/>
			</ode>
		</physics>


		<light type="directional" name="sun" cast_shadows="true">
			<origin pose="0 0 10 0 0 0"/>
			<diffuse rgba=".8 .8 .8 1"/>
			<specular rgba=".1 .1 .1 1"/>
			<attenuation range="10" constant="0.8" linear="0.01" quadratic="0.0"/>
			<direction xyz="0 .5 -.5"/>
		</light>

		<!-- Ground Plane -->
		<model name="plane_model" static="true">
			<link name="body">
				<collision name="geom">
					<geometry>
						<plane normal="0 0 1"/>
					</geometry>
					<surface>
						<friction>
							<ode mu="100000.0" mu2="100000.0"/>
						</friction>
					</surface>
				</collision>

				<visual name="visual" cast_shadows="false">
					<geometry>
						<plane normal="0 0 1"/>
					</geometry>
					<material script="Kitchen/Grass"/>
				</visual>
			</link>
		</model>

		<model name="model_1" static="false">
		<origin pose="0 0 0 0 0 0"/>

			<link name="link_1" gravity="1" self_collide="0">
				<origin pose="0 0 0.05 0 0 0"/>

				<inertial mass="1">
					<origin pose="0 0 0 0 0 0"/>
					<inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
				</inertial>
					
				<collision name="collision">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<surface>
						<friction>
							<ode mu='0.9' mu2='0.9' fdir1='0.0 0.0 0.0' slip1='0.0' slip2='0.0'/>
						</friction>
						<bounce restitution_coefficient='0.0' threshold='0.0'/>
						<contact>
							<ode soft_cfm='0.0' soft_erp='0.2' kp='10000000000000.0' kd='100000000000.0' max_vel='-1.0' min_depth='0.0'/>
						</contact>
					</surface>
				</collision>

				<visual name="visual" cast_shadows="true">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<material script="Gazebo/Blue"/>
				</visual>
			</link>

		</model>

		<model name="model_2" static="false">
		<origin pose="0 0 0 0 0 0"/>

			<link name="link_2" gravity="1" self_collide="0">
				<origin pose="0.2 0 0.05 0 0 0"/>

				<inertial mass="1">
					<origin pose="0 0 0 0 0 0"/>
					<inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
				</inertial>
					
				<collision name="collision">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<surface>
						<friction>
							<ode mu='1' mu2='1' fdir1='0.0 0.0 0.0' slip1='0.0' slip2='0.0'/>
						</friction>
						<bounce restitution_coefficient='0.0' threshold='0.0'/>
						<contact>
							<ode soft_cfm='0.0' soft_erp='0.2' kp='10000000000000.0' kd='100000000000.0' max_vel='-1.0' min_depth='0.0'/>
						</contact>
					</surface>
				</collision>

				<visual name="visual" cast_shadows="true">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<material script="Gazebo/Green"/>
				</visual>
			</link>

		</model>

		<model name="model_3" static="false">
		<origin pose="0 0 0 0 0 0"/>

			<link name="link_3" gravity="1" self_collide="0">
				<origin pose="0.4 0 0.05 0 0 0"/>

				<inertial mass="1">
					<origin pose="0 0 0 0 0 0"/>
					<inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
				</inertial>
					
				<collision name="collision">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<surface>
						<friction>
							<ode mu='1' mu2='1' fdir1='0.0 0.0 0.0' slip1='0.0' slip2='0.0'/>
						</friction>
						<bounce restitution_coefficient='0.0' threshold='0.0'/>
						<contact>
							<ode soft_cfm='0.0' soft_erp='0.2' kp='10000000000000.0' kd='100000000000.0' max_vel='-1.0' min_depth='0.0'/>
						</contact>
					</surface>
				</collision>

				<visual name="visual" cast_shadows="true">
					<origin pose="0 0 0 0 0 0"/>
					<geometry>
						<box size="0.05 0.3 0.05" />
					</geometry>
					<material script="Gazebo/Red"/>
				</visual>
			</link>

		</model>
	<plugin name="ground_contact" filename="libground_contact.so"/>

	</world>
</gazebo>

